### PR TITLE
Update .dockerignore (#180)

### PR DESCRIPTION
**Pull request by ssciolla**
This PR updates the repo's `.dockerignore` file to ensure unnecessary and sensitive files are not included in Docker images built locally. Changes include use of the `**/` Docker-specific syntax to properly find files with certain patterns in any directory; widening of the `env` pattern to capture HJSON configuration files; and the removal of the `.vscode` directory from the list (as it might be needed in our deployment environments). The PR aims to resolve issue #180.

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
+**/*.csv
 .data
 
 # etc
 .DS_Store
-*.pyc
-.vscode
-__pycache__
\ No newline at end of file
+**/*.pyc
```
Comment (jonespm) [inline .dockerignore:11]: So the .dockerignore has a different syntax from .gitignore? That's kind of annoying, thanks for noticing this!

File: .dockerignore [hunk 2]
```diff
@@ -1,13 +1,12 @@
...
 
 # data
-*.csv
+**/*.csv
 .data
 
 # etc
 .DS_Store
-*.pyc
-.vscode
```
Comment (jonespm) [inline .dockerignore:12]: I don't think there's any need to include this in the image but there's no harm in it either. You'd have to be running vscode inside the container for this to matter.

File: .dockerignore (hunk 1 above)
Comment (ssciolla) [reply .dockerignore:11]: Yeah, it's described near the bottom of this section: https://docs.docker.com/engine/reference/builder/#dockerignore-file

File: .dockerignore (hunk 2 above)
Comment (ssciolla) [reply .dockerignore:12]: I guess I was thinking about the debugger (which I still don't have a lot of experience with). You mentioned at one point that you can use this OpenShift. Would you not need `.vscode` to do that? I can add this back if it's not needed.

Comment (ssciolla) [reply .dockerignore:12]: Okay, that makes sense. I'll add it back to the file.

**Comment by ssciolla** [comment]
Thanks for the reviews! I'm merging now.

**Pull request merged**